packages/kit/src/core/dev/plugin.js

```diff
@@ -195,9 +195,15 @@ export async function create_plugin(config) {
 							const file = config.kit.files.assets + pathname;
 
 							if (fs.existsSync(file) && !fs.statSync(file).isDirectory()) {
-								req.url = encodeURI(pathname); // don't need query/hash
-								asset_server(req, res);
-								return;
+								const has_correct_case = fs
+									.readdirSync(path.dirname(file))
+									.includes(path.basename(file));
```

Comment:
How about using `realpath.native` instead?

```suggestion
								const has_correct_case = fs.realpathSync.native(file) === path.resolve(file);
```

Reply:
I started getting unexpected 404s after updating SvelteKit today. As it turns out, if you have a symlinked subdirectory in `static`, realpath and resolve _don't_ give you the same result.

Monkey-patching the check to use the initial version stopped the 404s, since the directory _does_ indeed contain a file by that name. You might want to double-check this fix.

(As a detail, this was happening on Windows, but it seems like it should also bite POSIX systems.)

(Also, considering this is merged and resolved, I should probably ensure @mrkishi and @Rich-Harris get pinged about this. 😃)